waiter/src/waiter/auth/authentication.clj

```diff
@@ -23,15 +23,19 @@
 (def ^:const AUTH-COOKIE-NAME "x-waiter-auth")
 
 (defprotocol Authenticator
+  (process-callback [this request scheme operation]
```

Comment:
The `scheme` and `operation` should come in as part of the request as route parameters.

Also, can we rename `scheme` to `auth-scheme` (`auth-protocol` if you rename scheme to protocol).

Reply:
I like to split it out earlier. don't have to do it two places

Reply:
changed